language/move-lang/tests/move_check/typing/return_type_last_exp_invalid.move

```diff
@@ -0,0 +1,19 @@
+module M {
+    resource struct R {}
+
+    t0(): u64 {
+        ()
+    }
+
+    t1(): () {
+        0
+    }
+
+    t2(): (u64, bool) {
+        (0, false, R{})
+    }
+
+    t3(): (u64, bool, R, bool) {
+        (0, false, R{})
+    }
+}
```

Comment:
Might also be worthwhile to have a test for same number of values but in the incorrect order to the type (e.g. return `(0, false, R)` to a return type of `(bool, u64, R)`).